test/src/test_encoded_value.cpp

```diff
@@ -105,6 +105,46 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(
   BOOST_CHECK_EQUAL(value_to_string(b), "nil");
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(
+    json_encoded_value_should_compare_for_equality,
+    encoded_value_type,
+    encoded_value_types) {
+  encoded_value_type a("null");
+  encoded_value_type b("1");
+  BOOST_CHECK(a == a);
+  BOOST_CHECK(!(a == b));
+  BOOST_CHECK(!(a != a));
+  BOOST_CHECK(a != b);
```

Comment:
I like to make it obvious that it's using the operators I want it to use here. It's not obvious that `BOOST_CHECK_NE` actually uses `operator!=`